### PR TITLE
Split sc.h into multiple headers.

### PR DESCRIPTION
**Pull request by dvander**
I don't know why every single thing is shoved into sc.h. This patch reduces the clutter a bit.

### CODE REVIEW

File: compiler/assembler.h [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// vim: set ts=8 sts=2 sw=2 tw=99 et:
+//
+//  Pawn compiler - Error message system
```
Comment (peace-maker) [inline compiler/assembler.h]: Every new header file has this comment about an `Error message system`. Might want to remove that / change it.

File: compiler/parser.cpp [hunk 2]
```diff
@@ -5428,10 +5436,6 @@ static void statement(int *lastindent,int allow_decl)
     doassert();
     lastst=tASSERT;
     break;
-  case tSLEEP:
```
Comment (peace-maker) [inline compiler/parser.cpp]: Did you mean to slip this removal into this commit?

Comment (dvander) [reply compiler/parser.cpp]: Yes, indeed.

File: compiler/assembler.h (hunk 1 above)
Comment (dvander) [reply compiler/assembler.h]: Thanks, I'll fix this up in the next commit.

**Pull request merged**